File: Easy/1331. Rank Transform of an Array/solution.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        int N = arr.size();

        vector<pair<int, int>> myVec;
        for (int i = 0; i < N; i++){
            myVec.push_back({arr[i], i});
        }

        sort(myVec.begin(), myVec.end());

        vector<int> result(N);
        int rank = 1;

        for (int i = 0; i < N; i++){
            if (i > 0 && myVec[i].first != myVec[i - 1].first){
                rank++;
            }

            result[myVec[i].second] = rank;
        }

        return result;
    }
};

File: Easy/1331. Rank Transform of an Array/solution.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        N = len(arr)

        myVec = []
        for i in range(N):
            myVec.append([arr[i], i])

        myVec.sort()

        result = [0] * N
        rank = 1

        for i in range(N):
            if i > 0 and myVec[i][0] != myVec[i - 1][0]:
                rank += 1

            result[myVec[i][1]] = rank

        return result